### PR TITLE
SPLICE-1357 Fix wrong result for right outer join that is performed through spark…

### PR DESCRIPTION
**Pull request by yxia92**
… engine - SPLICE-1357

### CODE REVIEW

File: hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSet.java [hunk 1]
```diff
@@ -638,8 +635,9 @@ private String planIfLast(AbstractSpliceFunction f, boolean isLast) {
                 expr = i!=0?expr.and(joinEquality):joinEquality;
             }
             DataSet joinedSet;
+
             if (op.wasRightOuterJoin)
-                joinedSet =  new SparkDataSet(rightDF.join(leftDF,expr,joinType.strategy()).rdd().toJavaRDD().map(
+                joinedSet =  new SparkDataSet(rightDF.join(leftDF,expr,"rightouter").rdd().toJavaRDD().map(
                         new RowToLocatedRowFunction(context)));
             else
```
Comment (jyuanca) [inline hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSet.java]: use joinType.RIGHTOUTER

Comment (yxia92) [inline hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSet.java]: I made the code change as you suggested. Thanks.

**Comment by jyuanca** [comment]
Moderator needs to squash and push

**Comment by snowspeeder** [comment]
do we need backport fix to 2.5, 2.0 branch?

**Comment by yxia92** [comment]
I can backport the fix to 2.5 and 2.0 too if required.

**Comment by wtang16** [comment]
Let's merge this to master first and run full regression and benchmarks before backporting it into 2.5 and 2.0.

**Pull request merged**